### PR TITLE
adding platform specific reference link to menu

### PR DESCRIPTION
**Pull request by katiegoines**
#### Description of changes:

staging: https://next-release-references.d1egzztxsxq9xz.amplifyapp.com/

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/directory/directory.mjs [hunk 1]
```diff
@@ -1103,11 +1103,51 @@ export const directory = {
           ]
         },
         {
-          path: 'src/pages/[platform]/reference/index.mdx',
-          children: [
-            { path: 'src/pages/[platform]/reference/flutter-api/index.mdx' }
+          isExternal: true,
+          route: 'https://aws-amplify.github.io/amplify-js/api/',
```
Comment (reesscot) [inline src/directory/directory.mjs:1107]: Think we should kill this as it's just relevant for JS v5, there is no and won't be a v6 version.

Comment (katiegoines) [reply src/directory/directory.mjs:1107]: We can but I've hidden it for v6 in this PR: https://next-release-references.d1egzztxsxq9xz.amplifyapp.com/

**Review by heatheramz — DISMISSED** [review]
LGTM

File: src/components/Menu/MenuItem.tsx [hunk 2]
```diff
@@ -97,6 +97,19 @@ export function MenuItem({
 
   const currentStyle = current ? 'menu__list-item__link--current' : '';
 
+  let hideAPIResources = false; 
+
+  if ((currentPlatform == 'javascript' ||
```
Comment (reesscot) [inline src/components/Menu/MenuItem.tsx:102]: I think we have a list of web platforms we could use here

Comment (katiegoines) [reply src/components/Menu/MenuItem.tsx]: Yes, yes there is. Fixed, thanks!

File: src/components/Menu/MenuItem.tsx [hunk 3]
```diff
@@ -121,7 +129,10 @@ export function MenuItem({
     return <></>;
   }
 
-  if (pageNode.isExternal) {
+  if (hideAPIResources) {
+    //do nothing
```
Comment (reesscot) [inline src/components/Menu/MenuItem.tsx:133]: how can we do nothing, don't we need to return jsx?

**Review by reesscot — APPROVED** [review]

**Review by heatheramz — APPROVED** [review]
LGTM

**Pull request merged**